scss/components/_button.scss

```diff
@@ -158,14 +158,17 @@ $button-transition: background-color 0.25s ease-out, color 0.25s ease-out !defau
 }
 
 /// Adds disabled styles to a button by fading the element, reseting the cursor, and disabling pointer events.
-@mixin button-disabled($color: $primary-color) {
+@mixin button-disabled(
+  $background: $primary-color,
+  $color: $button-color
+) {
   opacity: $button-opacity-disabled;
   cursor: not-allowed;
+  background-color: $background;
+  
+  $color: color-pick-contrast($background, ($button-color, $button-color-alt));
```

Comment:
If i do like this .... it breaks ...... the success and warning text color changes to white ....

```scss
@if $color == default {
    $color: color-pick-contrast($background, ($button-color, $button-color-alt));
  }
```